src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -447,6 +452,182 @@ public static AutoRepairHistory getHostWithLongestUnrepairTime(RepairType repair
+    }
+
+
+    /**
+     * @return Whether the host for the given eligibleRepairHistory has any replicas in hostsBeingRepaired.
+     * @param eligibleHistory History of node to check
+     * @param hostsBeingRepaired Hosts being repaired.
+     * @param myId Host id of this node, if the repair history is for this node, additional logging will take place.
+     * @param replicationStrategies Mapping of unique replication strategies to keyspaces having that strategy.
+     */
+    private static boolean hasReplicaWithOngoingRepair(AutoRepairHistory eligibleHistory,
+                                                       Map<InetAddressAndPort, UUID> hostsBeingRepaired,
+                                                       UUID myId,
+                                                       Map<AbstractReplicationStrategy, List<String>> replicationStrategies)
+    {
+        // If no broadcast address found for this host id in cluster metadata, just skip it, a node should always
+        // see itself in cluster metadata.
+        InetAddressAndPort eligibleBroadcastAddress = getBroadcastAddress(eligibleHistory.hostId);
+        if (eligibleBroadcastAddress == null)
+        {
+            return true;
+        }
+
+        // For each replication strategy, determine if host being repaired is a replica of the local node.
+        for (Map.Entry<AbstractReplicationStrategy, List<String>> entry : replicationStrategies.entrySet())
+        {
+            AbstractReplicationStrategy replicationStrategy = entry.getKey();
+            EndpointsByRange endpointsByRange = replicationStrategy.getRangeAddresses(ClusterMetadata.current());
+
+            // get ranges of the eligible address for the given replication strategy.
+            RangesAtEndpoint rangesAtEndpoint = StorageService.instance.getReplicas(replicationStrategy, eligibleBroadcastAddress);
+            for (Replica replica : rangesAtEndpoint)
+            {
+                // get the endpoints involved in this range.
+                EndpointsForRange endpointsForRange = endpointsByRange.get(replica.range());
+                // For each host in this range...
+                for (InetAddressAndPort inetAddressAndPort : endpointsForRange.endpoints())
+                {
+                    // If the address of the node in the range belongs to a host being repaired, return true.
+                    UUID hostId = hostsBeingRepaired.get(inetAddressAndPort);
+                    if (hostId != null)
+                    {
+                        // log if the repair history matches the current running node.
+                        InetAddressAndPort myBroadcastAddress = getBroadcastAddress(myId);
+                        if (myBroadcastAddress != null && myBroadcastAddress.equals(eligibleBroadcastAddress))
+                        {
+                            logger.info("Deferring repair because replica {} ({}) with shared ranges for " +
```

Comment:
Add a metric so we can know the starvation duration

Reply:
easy enough to add, could just update a timer that's a delta between the (now - lastRepairFinishTime - minRepairInterval) will do

Reply:
Added `RepairStartLagSec` for this